pkg/controller/kogitoruntime/deployer.go

```diff
@@ -73,13 +76,46 @@ func onGetComparators(comparator compare.ResourceComparator) {
 func onObjectsCreate(cli *client.Client, kogitoService v1alpha1.KogitoService) (resources map[reflect.Type][]resource.KubernetesResource, lists []runtime.Object, err error) {
 	resources = make(map[reflect.Type][]resource.KubernetesResource)
 
-	resObjectList, resType, res := createProtoBufConfigMap(kogitoService)
+	resObjectList, resType, res := createProtoBufConfigMap(cli, kogitoService)
 	lists = append(lists, resObjectList)
 	resources[resType] = []resource.KubernetesResource{res}
 	return
 }
 
-func createProtoBufConfigMap(kogitoService v1alpha1.KogitoService) (runtime.Object, reflect.Type, resource.KubernetesResource) {
+func getProtobufData(cli *client.Client, kogitoService v1alpha1.KogitoService) (map[string]string) {
+	// print endpoint
+	protobufEndpoint := kogitoService.GetStatus().GetExternalURI() + "/persistence/protobuf/"
```

Comment:
Hi @Kevin-Mok , I know this is still a draft but I was having a look since I have to do the same job for the dashboards. I think `kogitoService.GetStatus().GetExternalURI()` is not always populated. It's the case of kubernetes for instance, unless the user specifies it in the yaml

Reply:
Good point! You should use the `http://service.namespace` format instead. The services are always created and are accessible within the cluster.